src/main/java/gov/usgs/wma/waterdata/collections/CollectionsController.java

```diff
@@ -100,19 +108,18 @@ public String getOgcCollectionFeatures(
 			@RequestParam(value = "bbox", required = false) List<String> bbox,
 			@PathVariable(value = "collectionId") String collectionId, HttpServletResponse response) {
 
-		int count = collectionsDao.getCollectionFeatureCount(collectionsParams.buildParams(collectionId));
 
 		String rtn = null;
-		if (limit == 0 || startIndex >= count) {
-			response.setStatus(HttpStatus.NOT_FOUND.value());
-		} else {
-			rtn = collectionsDao.getCollectionFeaturesJson(
-					collectionsParams.buildParams(collectionId, limit, startIndex, bbox, count));
-			if (rtn == null) {
-				response.setStatus(HttpStatus.NOT_FOUND.value());
-			}
+		// requesting less than one is a non-starter
+		resultOr404(response, limit>0 ?"Good" :null);
+		if (response.getStatus() == HttpServletResponse.SC_OK) {
+			// check the collection feature count
+			int count = collectionsDao.getCollectionFeatureCount(collectionsParams.buildParams(collectionId));
+			// verify the start index is within the feature count
+			resultOr404(response, startIndex<count  ?"Good" :null);
```

Comment:
I understand the desire to be dry, but this is just confusing to me. (passing in Good or null)

Reply:
Yes, I thought about that. Have any ideas to make it work better. One thing I do when coding: if I cannot think of a better way to do something, get something that works and be open for refactor. I sent in null because that is what the all the others in the pattern are checking. This one is checking limits. I toyed with a method sig overload that took a value to compare.

I will look at it some more.

Reply:
I am just going to revert this.